Issue #4685 - jetty-slf4j-impl not a compile/runtime dependency.

joakime opened the pull request:
+ Mark as a test dependency

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - jetty-slf4j-impl not a compile/runtime dependency.

+ Mark as a test dependency

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - jetty-slf4j-impl not a compile/runtime dependency.

+ Mark as a test dependency

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

olamy commented on jetty-http-spi/pom.xml:
```diff
@@ -35,6 +35,16 @@
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <scope>provided</scope>
```
why provided here?

sbordet replied:
Ditto

joakime replied:
Historically, there's been no hard dependencies in this project.
See for yourself, there's no compile/runtime dependencies.

sbordet replied:
@joakime if there is a test failure and I want to debug it, why should I change the POM? I would just have `scope=test` like everywhere else.

joakime replied:
If there is a test failure and you want to debug it, don't change the pom.

On maven slf4j-api + jetty-slf4j-impl + jul-to-slf4j are all included in this project.
In your IDE slf4j-api + jetty-slf4j-impl + jul-to-slf4j  are all included in this project.

It's just not included when you use/depend on the `jetty-http-spi` artifact via maven, and like all of the other dependencies in `jetty-http-spi`, nothing is included, intentionally.

olamy reviewed (approved):
LGTM except the why provided?

sbordet commented on jetty-jmx/pom.xml:
```diff
@@ -42,6 +42,10 @@
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
```
I'm fine this the change above, but below `jetty-slf4j-impl` is both `provided` and `optional`.
I think it should be like elsewhere, in scope `test` if we use it in tests, or otherwise not even declared.

joakime replied:
Fixed

sbordet commented on jetty-websocket/websocket-javax-tests/pom.xml:
```diff
@@ -15,6 +15,10 @@
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
```
Not here, but below `jetty-slf4j-impl` should have `<scope>test</scope>` like everywhere else.

joakime replied:
Adjusted top level dependencyManagement for `jetty-slf4j-impl` to be `<scope>test</scope>` instead.

sbordet commented on tests/test-quickstart/pom.xml:
```diff
@@ -102,7 +102,14 @@
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-slf4j-impl</artifactId>
+    </dependency>
```
Both should have `scope=test`.

joakime replied:
`test-quickstart` has no test scoped dependencies currently, despite it consisting of 100% `src/test/java`.
guess I'll make all of it `<scope>test</scope>` then.

sbordet commented on tests/test-sessions/test-sessions-common/pom.xml:
```diff
@@ -29,10 +29,13 @@
+    </dependency>
     <dependency>
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-slf4j-impl</artifactId>
```
Must have `scope=test`

joakime replied:
I cleaned up all of the `/tests/test-sessions/test-*-sessions/` projects.
Improved the logging on infinispan and hazelcast testing in the process too.

sbordet reviewed (changes requested):
I also would like to see `jetty-slf4j-impl` be declared in the main POM's `dependencyManagement` with `scope=test` as a safety net. Only in the 2-3 places where we actually want it in a different scope, we explicitly declare so.

joakime committed:
Issue #4685 - Mark jetty-slf4j-impl as test scope by default

+ Update logging in test-hazelcast-sessions
+ Update logging in test-infinispan-sessions

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - jetty-http-spi logging for both jetty and jul

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - eliminate <version> in jetty-slf4j-impl

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime commented on jetty-websocket/websocket-javax-tests/pom.xml:
```diff
@@ -37,7 +41,6 @@
     <dependency>
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-slf4j-impl</artifactId>
```
This is a comment on jetty-slf4j-impl.

joakime committed:
Issue #4685 - Cleanup from review

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - Declare <scope> for jetty-slf4j-impl everywhere

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - Removing exclusion on jetty-slf4j-impl

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4685 - Adding jetty-slf4j-impl to test wars WEB-INF/lib/

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>
